logstash-core/src/main/java/org/logstash/ackedqueue/Batch.java

```diff
@@ -0,0 +1,44 @@
+package org.logstash.ackedqueue;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class Batch implements Closeable {
+
+    private final List<Queueable> elements;
+
+    private final List<Long> seqNums;
+    private final Queue queue;
+    private final AtomicBoolean closed;
+
+    public Batch(List<Queueable> elements, List<Long> seqNums, Queue q) {
+        this.elements = elements;
+        this.seqNums = seqNums;
+        this.queue = q;
+        this.closed = new AtomicBoolean(false);
+    }
+
+    // close acks the batch ackable events
+    public void close() throws IOException {
+        if (closed.getAndSet(true) == false) {
+              this.queue.ack(this.seqNums);
+        } else {
+            // TODO: how should we handle double-closing?
```

Comment:
it's a TODO. Undecided if we could just ignore double-closing or we should throw an exception. there's a P1 task to that effect. I think we can ignore this for now